cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -1790,6 +1792,167 @@ def test_new_storage_multiple_subnets_validator(queues, new_storage_count, failu
     assert_failure_level(actual_failures, failure_level)
 
 
+@pytest.mark.parametrize(
+    "queues, subnet_az_mappings, fsx_az_list, failure_level, expected_message",
+    [
+        (
+            [
+                SlurmQueue(
+                    name="single-az-different-subnet-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-2"],
+                    ),
+                ),
+                SlurmQueue(
+                    name="single-az-same-subnet-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-1"],
+                    ),
+                ),
+            ],
+            [{"subnet-2": "us-east-1b"}, {"subnet-1": "us-east-1a"}],
+            ["us-east-1a"],
+            FailureLevel.INFO,
+            "Your configuration for Queue 'single-az-different-subnet-queue' includes multiple subnets and external "
+            "shared storage configuration. Accessing a shared storage from different AZs can lead to increased "
+            "latency and costs.",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="same-az-same-subnet-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-1"],
+                    ),
+                ),
+                SlurmQueue(
+                    name="same-az-other-subnet-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-3"],
+                    ),
+                ),
+            ],
+            [{"subnet-1": "us-east-1a"}, {"subnet-3": "us-east-1a"}],
+            ["us-east-1a"],
+            None,
+            "",
+        ),
+        (
+            [
+                SlurmQueue(
+                    name="one-az-match-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-1"],
+                    ),
+                ),
+                SlurmQueue(
+                    name="full-az-match-queue",
+                    compute_resources=[],
+                    networking=SlurmQueueNetworking(
+                        subnet_ids=["subnet-1", "subnet-2"],
+                    ),
+                ),
+            ],
+            [{"subnet-1": "us-east-1a"}, {"subnet-1": "us-east-1a", "subnet-2": "us-east-1b"}],
+            ["us-east-1a", "us-east-1b"],
+            FailureLevel.INFO,
+            "Your configuration for Queue 'one-az-match-queue' includes multiple subnets and external "
```

Comment:
there should be no error in this case.
FS is in AZ ["us-east-1a", "us-east-1b"], so cluster will be able to mount the FS in the same AZ

Reply:
This is true IF and OLNY IF the data in each AZ is an exact replica and so a compute node in `us-east-1a` will connect to the FS mount point in `us-east-1a` and all the data will be local.

If, somehow, this is not true data will be transferred from one AZ to another and this will have an impact on the latency while accessing the data and possibly cost.

Reply:
We discussed this topic in a call a decided to slightly change the logic of the validator.